### ISSUE TITLE
Error on running image

### ISSUE THREAD

**Issue by diogodias24** [democratic-csi/democratic-csi#171]
Hi, I don't know if this is the right place for opening this bug, but here it goes.

Since today after re downloading the latest image of:

`docker.io/democraticcsi/democratic-csi:latest`

I just had a failing deployment, container is throwing:

`/usr/bin/env: only ${VARNAME} expansion is supported, error at: ${NO`

On docker hub just saw your image was updated yesterday, so maybe a docker file error?
Can you point me on the right direction for maybe helping?

**Comment by travisghansen** [comment]
Hmm, let me take a peek.

**Comment by travisghansen** [comment]
Which architecture is the node/cluster? I've been running some tests here with amd64 and everything seems clear..

**Comment by diogodias24** [comment]
> Which architecture is the node/cluster? I've been running some tests here with amd64 and everything seems clear..

Debian AMD64
1 Control Plane Kubeadm v1.23.5 
2 Nodes v1.23.5 

Do you need more debug/info?

**Comment by travisghansen** [comment]
Can you share the exact image tag/hash being used?

**Comment by diogodias24** [comment]
> Can you share the exact image tag/hash being used?

Sure! Right now with:

`sha256:06aad82e9aada8d2feacfbf0b229e7d6647b69a03397591d241f1c2b3f3cd7d8`

Already tried with `v.1.6.2` as well:

`sha256:8b9d3474c427969bd3fd44c65fd90b544d47f51d1b9040f80df9e295ae24fd0a`

**Comment by travisghansen** [comment]
Are you using a proxy/mirror or something? Those hashes don't match anything that I've found.

- https://hub.docker.com/layers/democraticcsi/democratic-csi/latest/images/sha256-0f6f618c8e6f9aaefcb0972a50811399d2a00bdfa90de02c56b9549f0b0137f2?context=explore
- https://hub.docker.com/layers/democraticcsi/democratic-csi/v1.6.2/images/sha256-2fa0d8e70fa6093dc949eb9c946e1f683a5b495df98e4541b5874e966d476214?context=explore

If I try to pull those hashes locally it errors out (as in they don't exist at all).

```
docker pull democraticcsi/democratic-csisha256:06aad82e9aada8d2feacfbf0b229e7d6647b69a03397591d241f1c2b3f3cd7d8


Error response from daemon: manifest for democraticcsi/democratic-csisha256:06aad82e9aada8d2feacfbf0b229e7d6647b69a03397591d241f1c2b3f3cd7d8 not found: manifest unknown: manifest unknown

docker pull democraticcsi/democratic-csi@sha256:8b9d3474c427969bd3fd44c65fd90b544d47f51d1b9040f80df9e295ae24fd0a
Error response from daemon: received unexpected HTTP status: 500 Internal Server Error
```

**Comment by diogodias24** [comment]
> Are you using a proxy/mirror or something? Those hashes don't match anything that I've found.
> 
> * https://hub.docker.com/layers/democraticcsi/democratic-csi/latest/images/sha256-0f6f618c8e6f9aaefcb0972a50811399d2a00bdfa90de02c56b9549f0b0137f2?context=explore
> * https://hub.docker.com/layers/democraticcsi/democratic-csi/v1.6.2/images/sha256-2fa0d8e70fa6093dc949eb9c946e1f683a5b495df98e4541b5874e966d476214?context=explore
> 
> If I try to pull those hashes locally it errors out (as in they don't exist at all).
> 
> ```
> docker pull democraticcsi/democratic-csisha256:06aad82e9aada8d2feacfbf0b229e7d6647b69a03397591d241f1c2b3f3cd7d8
> Error response from daemon: manifest for democraticcsi/democratic-csisha256:06aad82e9aada8d2feacfbf0b229e7d6647b69a03397591d241f1c2b3f3cd7d8 not found: manifest unknown: manifest unknown
> 
> docker pull democraticcsi/democratic-csi@sha256:8b9d3474c427969bd3fd44c65fd90b544d47f51d1b9040f80df9e295ae24fd0a
> Error response from daemon: received unexpected HTTP status: 500 Internal Server Error
> ```

You are right, I'm sorry misunderstood and gave the wrong id's.
So I just did `docker system prune -a` and pulled again. Now using:

`democraticcsi/democratic-csi@sha256:d4545813c04a9942a221e67c37207359431a4bf3031339cd03699537d8751c10`

## Worker:

<img width="1086" alt="worker" src="https://user-images.githubusercontent.com/35083226/162585502-83f467fb-11ef-4ace-815f-8402b7493b2b.png">

## Tried the same hash on raspberry (arm64) and have a different behaviour:

<img width="1065" alt="pi" src="https://user-images.githubusercontent.com/35083226/162585506-aa71a3fb-8c56-4645-9006-8e89a742c212.png">

**Comment by travisghansen** [comment]
So weird..

```
# from amd64 machine..
docker run --rm -it democraticcsi/democratic-csi@sha256:d4545813c04a9942a221e67c37207359431a4bf3031339cd03699537d8751c10


Unable to find image 'democraticcsi/democratic-csi@sha256:d4545813c04a9942a221e67c37207359431a4bf3031339cd03699537d8751c10' locally
docker.io/democraticcsi/democratic-csi@sha256:d4545813c04a9942a221e67c37207359431a4bf3031339cd03699537d8751c10: Pulling from democraticcsi/democratic-csi
Digest: sha256:d4545813c04a9942a221e67c37207359431a4bf3031339cd03699537d8751c10
Status: Downloaded newer image for democraticcsi/democratic-csi@sha256:d4545813c04a9942a221e67c37207359431a4bf3031339cd03699537d8751c10
grpc implementation: @grpc/grpc-js
democratic-csi [options]

Options:
  --driver-config-file              provide a path to driver config file
                                                                      [required]
  --log-level                       log level
                 [choices: "error", "warn", "info", "verbose", "debug", "silly"]
  --csi-version                     versin of the csi spec to load
      [required] [choices: "0.2.0", "0.3.0", "1.0.0", "1.1.0", "1.2.0", "1.3.0",
                                                               "1.4.0", "1.5.0"]
  --csi-name                        name to use for driver registration
                                                                      [required]
  --csi-mode                        mode of the controller
                    [array] [required] [choices: "controller", "node"] [default:
                                                          ["controller","node"]]
  --server-address                  listen address for the server       [string]
  --server-port                     listen port for the server          [number]
  --server-socket                   listen socket for the server        [string]
  --server-socket-permissions-mode  permissions on the socket file for the
                                    server            [string] [default: "0600"]
  --version                         Show version number                [boolean]
  --help                            Show help                          [boolean]

Missing required arguments: driver-config-file, csi-version, csi-name
driver-config-file is required
csi-version is required
csi-name is required
```

Executing like that is indeed expected outcome (the missing required args).

Can you try the same command from another node/workstation? Interested why that node is behaving that way..

**Comment by travisghansen** [comment]
Where did you get that hash from?

**Comment by diogodias24** [comment]
> Where did you get that hash from?

This the latest. I just did a docker inspect to the image.

**Comment by diogodias24** [comment]
I can confirm this is a issue only on my Debian Instances. Just spin out a Ubuntu one, and it's working fine.

<img width="1175" alt="ubuntu_vs_debian" src="https://user-images.githubusercontent.com/35083226/162586714-32d632f7-8b88-45bf-bc1c-ac665557c0be.png">

## Working one:
```
NAME="Ubuntu"
VERSION="20.04.4 LTS (Focal Fossa)"
ID=ubuntu
ID_LIKE=debian
PRETTY_NAME="Ubuntu 20.04.4 LTS"
VE RSION_ID="20.04"
HOME_URL="https://www.ubuntu.com/"
SUPPORT_URL="https://help.ubuntu.com/"
BUG_REPORT_URL="https://bugs.launchpad.net/ubuntu/"
PRIVACY_POLICY_URL="https://www.ubuntu.com/legal/terms-and-policies/privacy-policy"
VERSION_CODENAME=focal
UBUNTU_CODENAME=focal
```


## Not working ones (2 Nodes):

```
PRETTY_NAME="Debian GNU/Linux 10 (buster)"
NAME="Debian GNU/Linux"
VERSION_ID="10"
VERSION="10 (buster)"
VERSION_CODENAME=buster
ID=debian
HOME_URL="https://www.debian.org/"
SUPPORT_URL="https://www.debian.org/support"
BUG_REPORT_URL="https://bugs.debian.org/"
```

I can send kernel and so if you want. But this is really strange.

**Comment by travisghansen** [comment]
The sha for latest (amd64) should be `sha256:0f6f618c8e6f9aaefcb0972a50811399d2a00bdfa90de02c56b9549f0b0137f2`. Can you try that?

For example:

```
docker run --rm -it --entrypoint bash democraticcsi/democratic-csi@sha256:0f6f618c8e6f9aaefcb0972a50811399d2a00bdfa90de02c56b9549f0b0137f2

env --version
./bin/democratic-csi
```

**Comment by diogodias24** [comment]
> The sha for latest (amd64) should be `sha256:0f6f618c8e6f9aaefcb0972a50811399d2a00bdfa90de02c56b9549f0b0137f2`. Can you try that?
> 
> For example:
> 
> ```
> docker run --rm -it --entrypoint bash democraticcsi/democratic-csi@sha256:0f6f618c8e6f9aaefcb0972a50811399d2a00bdfa90de02c56b9549f0b0137f2
> 
> env --version
> ./bin/democratic-csi
> ```

Here's my response:

```
 diogodias@debian-worker-1:~$ docker run --rm -it --entrypoint bash democraticcsi/democratic-csi@sha256:0f6f618c8e6f9aaefcb0972a50811399d2a00bdfa90de02c56b9549f0b0137f2
Unable to find image 'democraticcsi/democratic-csi@sha256:0f6f618c8e6f9aaefcb0972a50811399d2a00bdfa90de02c56b9549f0b0137f2' locally

docker.io/democraticcsi/democratic-csi@sha256:0f6f618c8e6f9aaefcb0972a50811399d2a00bdfa90de02c56b9549f0b0137f2: Pulling from democraticcsi/democratic-csi
Digest: sha256:0f6f618c8e6f9aaefcb0972a50811399d2a00bdfa90de02c56b9549f0b0137f2
Status: Downloaded newer image for democraticcsi/democratic-csi@sha256:0f6f618c8e6f9aaefcb0972a50811399d2a00bdfa90de02c56b9549f0b0137f2

root@a4c72abbf6b2:/home/csi/app#
root@a4c72abbf6b2:/home/csi/app# env --version
env (GNU coreutils) 8.32
Copyright (C) 2020 Free Software Foundation, Inc.
License GPLv3+: GNU GPL version 3 or later <https://gnu.org/licenses/gpl.html>.
This is free software: you are free to change and redistribute it.
There is NO WARRANTY, to the extent permitted by law.

Written by Richard Mlynarik, David MacKenzie, and Assaf Gordon.
root@a4c72abbf6b2:/home/csi/app# ./bin/democratic-csi
/usr/bin/env: only ${VARNAME} expansion is supported, error at: ${NO
root@a4c72abbf6b2:/home/csi/app#
```

**Comment by travisghansen** [comment]
Wondering if debian has something very strict for this: https://stackoverflow.com/questions/10813538/shebang-line-limit-in-bash-and-linux-kernel

If you exec into the container (using the previously sent command) and then install vi/vim you can test this theory by trying to execute `./bin/democratic-csi`.

The shebang line looks like this: https://github.com/democratic-csi/democratic-csi/blob/master/bin/democratic-csi#L1

You can safely remove `${NODE_OPTIONS_CSI_1} ${NODE_OPTIONS_CSI_2} ${NODE_OPTIONS_CSI_3} ${NODE_OPTIONS_CSI_4} ${NODE_OPTIONS_CSI_5}` and then try to launch again and see if it works.

**Comment by fumoretti** [comment]
Hi, just found recently the same situation.

Democratic-csi constantly crashing during deploy with only this on node pod log:

/usr/bin/env: only ${VARNAME} expansion is supported, error at: ${NO

Based on this issue discussion, I tried to run a container manualy testing different versions and found this error on latest, from v1.6.1 and before the problem not occur:

With latest (1.6.2 and above):

root@k8s-hml-node05:~# docker run --rm -ti democraticcsi/democratic-csi:latest --csi-version=1.5.0 --csi-name=org.democratic-csi.truenasnfs --driver-config-file=/config/driver-config-file.yaml --log-level=info --csi-mode=controller --server-socket=/csi-data/csi.sock.internal
/usr/bin/env: only ${VARNAME} expansion is supported, error at: ${NO

With 1.6.1:

root@k8s-hml-node05:~# docker run --rm -ti democraticcsi/democratic-csi:v1.6.1 --csi-version=1.5.0 --csi-name=org.democratic-csi.truenasnfs --driver-config-file=/config/driver-config-file.yaml --log-level=info --csi-mode=controller --server-socket=/csi-data/csi.sock.internal
grpc implementation: @grpc/grpc-js
democratic-csi [options]

Options:
  --driver-config-file  provide a path to driver config file          [required]
  --log-level           log level
                 [choices: "error", "warn", "info", "verbose", "debug", "silly"]
  --csi-version         versin of the csi spec to load
      [required] [choices: "0.2.0", "0.3.0", "1.0.0", "1.1.0", "1.2.0", "1.3.0",
                                                               "1.4.0", "1.5.0"]
  --csi-name            name to use for driver registration           [required]
  --csi-mode            mode of the controller
                    [array] [required] [choices: "controller", "node"] [default:
                                                          ["controller","node"]]
  --server-address      listen address for the server                   [string]
  --server-port         listen port for the server                      [number]
  --server-socket       listen socket for the server                    [string]
  --version             Show version number                            [boolean]
  --help                Show help                                      [boolean]

failed parsing config file: /config/driver-config-file.yaml


I think this problem is related with my S.O version (ubuntu 18.04 with 4.15.0-175-generic kernel) since with ubuntu 20.04 and same docker version the problem not exist.

I tried some ways to set the 1.6.1 version of democratic-csi during helm upgrade (--set=image.tag=v1.6.1) but the latest aways is used.

I need some help to force the 1.6.1 version or workaround this problem.. this is a non production kubernetes cluster, but i have a production one that probably will stop if any node reboot and redeploy democratic-csi (exactly same S.O and kernels on prod cluster).

**Comment by fumoretti** [comment]
> Wondering if debian has something very strict for this: https://stackoverflow.com/questions/10813538/shebang-line-limit-in-bash-and-linux-kernel
> 
> If you exec into the container (using the previously sent command) and then install vi/vim you can test this theory by trying to execute `./bin/democratic-csi`.
> 
> The shebang line looks like this: https://github.com/democratic-csi/democratic-csi/blob/master/bin/democratic-csi#L1
> 
> You can safely remove `${NODE_OPTIONS_CSI_1} ${NODE_OPTIONS_CSI_2} ${NODE_OPTIONS_CSI_3} ${NODE_OPTIONS_CSI_4} ${NODE_OPTIONS_CSI_5}` and then try to launch again and see if it works.

Just tried this. In fact, removing ${NODE_OPTIONS_CSI_1} ${NODE_OPTIONS_CSI_2} ${NODE_OPTIONS_CSI_3} ${NODE_OPTIONS_CSI_4} ${NODE_OPTIONS_CSI_5} from democratic-csi just works:

root@1a5b9534388c:/home/csi/app/bin# head -1 democratic-csi
#!/usr/bin/env -S node --expose-gc
root@1a5b9534388c:/home/csi/app/bin# ./democratic-csi
grpc implementation: @grpc/grpc-js
democratic-csi [options]

Options:
  --driver-config-file              provide a path to driver config file
                                                                      [required]
  --log-level                       log level
                 [choices: "error", "warn", "info", "verbose", "debug", "silly"]
  --csi-version                     versin of the csi spec to load
      [required] [choices: "0.2.0", "0.3.0", "1.0.0", "1.1.0", "1.2.0", "1.3.0",
                                                               "1.4.0", "1.5.0"]
  --csi-name                        name to use for driver registration
                                                                      [required]
  --csi-mode                        mode of the controller
                    [array] [required] [choices: "controller", "node"] [default:
                                                          ["controller","node"]]
  --server-address                  listen address for the server       [string]
  --server-port                     listen port for the server          [number]
  --server-socket                   listen socket for the server        [string]
  --server-socket-permissions-mode  permissions on the socket file for the
                                    server            [string] [default: "0600"]
  --version                         Show version number                [boolean]
  --help                            Show help                          [boolean]

Missing required arguments: driver-config-file, csi-version, csi-name
driver-config-file is required
csi-version is required
csi-name is required
root@1a5b9534388c:/home/csi/app/bin#

With v1.6.1 the first line of democratic-csi:

root@0323946fb541:/home/csi/app/bin# head -1 democratic-csi
#!/usr/bin/env -S node --nouse-idle-notification --expose-gc
root@0323946fb541:/home/csi/app/bin#

So, how i can force helm to use 1.6.1 of democratic-csi, at least until i can further investigate and maybe upgrade my workers nodes to ubuntu 20 (or newest kernel from mainline, etc...).

**Comment by diogodias24** [comment]
Sorry @travisghansen , didn't have the time to come back with the required debug info.

Nice job @fumoretti .
How are you managing the upgrades of the Chart?
For example, if you using Lens, you can do something like this:

<img width="721" alt="Screenshot 2022-04-14 at 17 07 32" src="https://user-images.githubusercontent.com/35083226/163430028-ae2870c5-2b71-4b09-9e29-892cb331f98c.png">

**Comment by fumoretti** [comment]
> Sorry @travisghansen , didn't have the time to come back with the required debug info.
> 
> Nice job @fumoretti . How are you managing the upgrades of the Chart? For example, if you using Lens, you can do something like this:
> 
> <img alt="Screenshot 2022-04-14 at 17 07 32" width="721" src="https://user-images.githubusercontent.com/35083226/163430028-ae2870c5-2b71-4b09-9e29-892cb331f98c.png">
> 
> And change the required image tag.
> 
> Hope that helps

Hello Diogo,

I'm using helm cli to do deploy and upgrade of democratic-csi. I have rancher too and tried an upgrade from it, editing the chart yaml and even that way the latest tag are deployed.

From helm cli i tried to upgrade using --set=image.tag=v1.6.1 with no good results... still deploying latest.

**Comment by fumoretti** [comment]
Diogo, forget my last message. I just tried again the upgrade using rancher UI, editing helm options and works well. I really don't know what I did wrong the last time i did this.

![democratic-csi-1 6 1-ok](https://user-images.githubusercontent.com/103659894/163433182-b58b470b-e564-42cb-8b9c-5e1922e95958.png)

**Comment by diogodias24** [comment]
> Diogo, forget my last message. I just tried again the upgrade using rancher UI, editing helm options and works well. I really don't know what I did wrong the last time i did this.
> 
> ![democratic-csi-1 6 1-ok](https://user-images.githubusercontent.com/103659894/163433182-b58b470b-e564-42cb-8b9c-5e1922e95958.png)

I think I had the same issue. It only worked for me on the cli when I removed the Chart and installed it again with that var on the command.
After that it just worked even if you set it up as latest.

**Comment by fumoretti** [comment]
Can you show more details on how you did the upgrade from cli with latest ? are you using Debian 10 or upgraded all workers to Ubuntu 20 ?

I think don't get what "installed it again with that var" means.

**Comment by fumoretti** [comment]
i did kernel upgrade on my non production cluster nodes (workers only), from 4.15 to 5.4-hwe and now all works fine, even in latest version of democratic-csi.

Thanks @diogodias24 and @travisghansen for this great support. You guys rock.

Now i'll plan to do the same on production nodes when possible (or rollout workers with ubuntu 20 lts nodes...)

**Comment by diogodias24** [comment]
> Can you show more details on how you did the upgrade from cli with latest ? are you using Debian 10 or upgraded all workers to Ubuntu 20 ?
> 
> I think i don't get what "installed it again with that var" means.

Nevermind, I just review and what I've done was downgrading the chart version. It didn't work because I think all the tag images by default  come with ":latest".

So since I can replace easily the nodes, I just switched the "workers" to ubuntu.

**Comment by travisghansen** [comment]
It sounds more and more like what I mentioned. I added those env vars in the shebang line to help debugging some other stuff but they aren’t critical so I can remove them. However I’d be interested to know if it works with 1 or 2 of them etc.

I’ve been super busy building out windows support and have made really good progress on that. After finishing that up I’ll come back to this and we’ll get it addressed.

**Comment by fumoretti** [comment]
> docker run --rm -it --entrypoint bash democraticcsi/democratic-csi@sha256:0f6f618c8e6f9aaefcb0972a50811399d2a00bdfa90de02c56b9549f0b0137f2

Hello Trevis,

I did a test, removing just one is enought to run properly:

root@fd307d8f1abc:/home/csi/app/bin# ./democratic-csi
grpc implementation: @grpc/grpc-js
democratic-csi [options]

Options:
  --driver-config-file              provide a path to driver config file
                                                                      [required]
  --log-level                       log level
                 [choices: "error", "warn", "info", "verbose", "debug", "silly"]
  --csi-version                     versin of the csi spec to load
      [required] [choices: "0.2.0", "0.3.0", "1.0.0", "1.1.0", "1.2.0", "1.3.0",
                                                               "1.4.0", "1.5.0"]
  --csi-name                        name to use for driver registration
                                                                      [required]
  --csi-mode                        mode of the controller
                    [array] [required] [choices: "controller", "node"] [default:
                                                          ["controller","node"]]
  --server-address                  listen address for the server       [string]
  --server-port                     listen port for the server          [number]
  --server-socket                   listen socket for the server        [string]
  --server-socket-permissions-mode  permissions on the socket file for the
                                    server            [string] [default: "0600"]
  --version                         Show version number                [boolean]
  --help                            Show help                          [boolean]

Missing required arguments: driver-config-file, csi-version, csi-name
driver-config-file is required
csi-version is required
csi-name is required

root@fd307d8f1abc:/home/csi/app/bin# head -1 democratic-csi
#!/usr/bin/env -S node --expose-gc ${NODE_OPTIONS_CSI_1} ${NODE_OPTIONS_CSI_2} ${NODE_OPTIONS_CSI_3} ${NODE_OPTIONS_CSI_4}


With all 5 the problem is triggered:

root@fd307d8f1abc:/home/csi/app/bin# head -1 democratic-csi 
#!/usr/bin/env -S node --expose-gc ${NODE_OPTIONS_CSI_1} ${NODE_OPTIONS_CSI_2} ${NODE_OPTIONS_CSI_3} ${NODE_OPTIONS_CSI_4} ${NODE_OPTIONS_CSI_5}
root@fd307d8f1abc:/home/csi/app/bin# ./democratic-csi
/usr/bin/env: only ${VARNAME} expansion is supported, error at: ${NO
root@fd307d8f1abc:/home/csi/app/bin#

Node server infos:

root@democratic-csi-tests:~# docker version
Client:
 Version:           20.10.7
 API version:       1.41
 Go version:        go1.13.8
 Git commit:        20.10.7-0ubuntu5~18.04.3
 Built:             Mon Nov  1 01:04:14 2021
 OS/Arch:           linux/amd64
 Context:           default
 Experimental:      true

Server:
 Engine:
  Version:          20.10.7
  API version:      1.41 (minimum version 1.12)
  Go version:       go1.13.8
  Git commit:       20.10.7-0ubuntu5~18.04.3
  Built:            Fri Oct 22 00:57:37 2021
  OS/Arch:          linux/amd64
  Experimental:     false
 containerd:
  Version:          1.5.5-0ubuntu3~18.04.2
  GitCommit:        
 runc:
  Version:          1.0.1-0ubuntu2~18.04.1
  GitCommit:        
 docker-init:
  Version:          0.19.0
  GitCommit:        
root@democratic-csi-tests:~# uname -a
Linux democratic-csi-tests 4.15.0-175-generic #184-Ubuntu SMP Thu Mar 24 17:48:36 UTC 2022 x86_64 x86_64 x86_64 GNU/Linux
root@democratic-csi-tests:~# lscpu 
Architecture:        x86_64
CPU op-mode(s):      32-bit, 64-bit
Byte Order:          Little Endian
CPU(s):              2
On-line CPU(s) list: 0,1
Thread(s) per core:  1
Core(s) per socket:  1
Socket(s):           2
NUMA node(s):        1
Vendor ID:           AuthenticAMD
CPU family:          25
Model:               1
Model name:          AMD EPYC 7713 64-Core Processor
Stepping:            1
CPU MHz:             1996.250
BogoMIPS:            3992.50
Hypervisor vendor:   VMware
Virtualization type: full
L1d cache:           32K
L1i cache:           32K
L2 cache:            512K
L3 cache:            262144K
NUMA node0 CPU(s):   0,1
Flags:               fpu vme de pse tsc msr pae mce cx8 apic sep mtrr pge mca cmov pat pse36 clflush mmx fxsr sse sse2 syscall nx mmxext fxsr_opt pdpe1gb rdtscp lm constant_tsc rep_good nopl tsc_reliable nonstop_tsc cpuid extd_apicid pni pclmulqdq ssse3 fma cx16 pcid sse4_1 sse4_2 x2apic movbe popcnt aes xsave avx f16c rdrand hypervisor lahf_lm extapic cr8_legacy abm sse4a misalignsse 3dnowprefetch osvw invpcid_single ibpb vmmcall fsgsbase bmi1 avx2 smep bmi2 invpcid rdseed adx smap clflushopt clwb sha_ni xsaveopt xsavec xsaves clzero arat pku ospke overflow_recov succor
root@democratic-csi-tests:~#

**Comment by travisghansen** [comment]
Very interesting, don't think I've ever seen that limit hit in practice. The env vars are for very advanced use-cases anyhow (and not meant to be used generally) so I'll fix it up shortly for the upcoming 1.7.0 release (a massive release adding support for windows nodes).

**Comment by travisghansen** [comment]
For future reference, it appears the limit you are hitting is 128 chars, so the shebang line needs to < 128.

**Comment by travisghansen** [comment]
Try with the most recent `next` image.

**Comment by diogodias24** [comment]
> Try with the most recent `next` image.

Let us know when you deliver a new version, and I can test with the old nodes, since I still have those on promox.

Thanks :)

**Comment by travisghansen** [comment]
I was hoping to get it tested before snapping a new release (chicken/egg I guess)..

**Comment by diogodias24** [comment]
Ok, If you want, maybe release a version with a different tag?
Let me know if you want to try something :)

**Comment by travisghansen** [comment]
Yeah, use the `next` tag…

**Comment by lawrencegripper** [comment]
Came here to say a massive thank you, this fixed the issue for me too :heart:

**Comment by travisghansen** [comment]
I’m still a bit out from a release so I’ll leave this open until it’s in a versioned release.

**Comment by travisghansen** [comment]
Fixed in commit: https://github.com/democratic-csi/democratic-csi/commit/6cde0d3a700a1218e293f09eaf700a43e42a6493

Released in `v1.7.0`
